### PR TITLE
Plugins: Fix plugin catalog filtering

### LINKED ISSUE #66478: Type filter doesn't work when combined with string search
**What happened**:

Issue 1: Originally filtered list of plugins by type=panel, then tried a search on name, but it removes the filter on type.

Issue 2: With a name search already active, attempting to apply a filter by type has no effect including on refresh.

https://user-images.githubusercontent.com/19860021/231797042-14b195eb-4a62-4fbd-b40c-021faf2f0a0c.mp4

**What you expected to happen**:

User should be able to combine both filters

**How to reproduce it (as minimally and precisely as possible)**:

1. Go to Administration > Plugins
2. Select Type: Panel
3. Search (Name): "Cal" (just as this brings up both datasources and panels)
4. See Type filter has been removed
5. Select Type: Panel again
6. See no difference

**Anything else we need to know?**:

**Environment**:
- Grafana version: Tested on HG 9.5 and locally 9.4.3 OSS
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser: MacOS / Chrome
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by leventebalogh**
Fixes https://github.com/grafana/grafana/issues/66478

### What changed?
Updated the filtering in the plugins catalog to be able to use the different filters together. Also added some tests for the `find()` selector.

https://user-images.githubusercontent.com/9974811/232472960-174b6c01-de07-4385-955a-d1a5f9c4c61f.mov

### CODE REVIEW

**Review by andresmgot — APPROVED** [review]
LGTM! I suggest adding another test but don't need to block this for that

**Review by sympatheticmoose — APPROVED** [review]
LGTM! Tested locally and behaves as expected now 🥇

File: public/app/features/plugins/admin/pages/Browse.tsx [hunk 1]
```diff
@@ -55,7 +55,7 @@ export default function Browse({ route }: GrafanaRouteComponentProps): ReactElem
   };
 
   const onSearch = (q: string) => {
```
[the remark this answers is not in the document]
Comment (leventebalogh) [reply public/app/features/plugins/admin/pages/Browse.tsx:57]: Yepp, I think that's a good point, will add another test case in the afternoon 👍

File: public/app/features/plugins/admin/components/SearchField.tsx [hunk 2]
```diff
@@ -20,6 +20,7 @@ const useDebounceWithoutFirstRender = (callBack: () => any, delay = 0, deps: Rea
         isFirstRender.current = false;
         return;
       }
+      console.log('--------- DEBUOUNCE');
```
Comment (andresmgot) [inline public/app/features/plugins/admin/components/SearchField.tsx:23]: I think you forgot to remove these @leventebalogh :sweat_smile:

**Pull request merged**